contributed/ra_dec_dc2_run2.2i_dask.ipynb

```diff
@@ -0,0 +1,1455 @@
+    "\n",
+    "And, yes, the calculation itself that we're doing can probably be better optimized to allow numpy to figure out what are the array operations.  But the more general point is that it's easy to have functions that work fine on 1-4 threads.  Diving in to the full details of what in `numpy` is parallelized across how many threads threads to what level in different backends is a rich explorational experience.  But the overhead to use multiple processes to do this in parallel is generally small; Dask is one easy way to build and schedule the work graph to do this work."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "%%timeit\n",
+    "x, y = moll_ang2xy(ddf['ra'], ddf['dec'], mollweide=mollweide)\n",
+    "print(x.mean(), y.mean())  # Trigger the computation"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If we do this all just in straight Pandas we would do:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%timeit\n",
+    "df = pd.read_parquet(data_path, columns=columns, engine='pyarrow')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The read is fast enough because `pyarrow` automatically uses the available CPUs to do parallel reads.  It looks up how many available cores it has with `pyarrow.cpu_count()` and uses all of them.\n",
+    "\n",
+    "However, if either `OMP_NUM_THREADS` or `OMP_THREADS_LIMIT` is set, it will use that value.  So if you are already set up to be in a multi-processing setup, you might find that these have been already set to as low as 1 and this `pandas.read_parquet` may in fact be somewhat slower."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pyarrow\n",
+    "pyarrow.cpu_count()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "And now we have to actually run the command.  The %%timeit magic prevents any of the directly set variables to outside its context."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df = pd.read_parquet(data_path, columns=columns, engine='pyarrow')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's try the projection calculation:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%timeit\n",
+    "df['x'], df['y'] = mollweide.ang2xy(df['ra'], df['dec'], lonlat=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Ouch!  I got 48 seconds on my 3GHz 8-core Xeon E5 desktop.\n",
+    "\n",
```

Comment:
At NERSC this was 1 min 11 seconds. Since you used `%%timeit` instead of `%%time`, it did that 7 times. Ouch indeed!  Because of this and all the following similar timing cases it takes a *really* long time to run through the notebook.  Consider using just `%%time`.